### PR TITLE
Feauture: Adding Stable-Video-Diffusion model for image2video generation.

### PR DESCRIPTION
**Pull request by jiexiong2016**
WIP

### CODE REVIEW

File: examples/stable-video-diffusion/models/svd.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        self,
+        image: Union[np.ndarray, Image.Image, str],
+        seed: int = -1,
+    ) -> Iterable[Image.Image]:
+        
+        if isinstance(image, np.ndarray):
+            image = Image.fromarray(image)
+
+        if isinstance(image, str):
+            if image.startswith("http://") or image.startswith("https://"):
```
Comment (spillai) [inline examples/stable-video-diffusion/models/svd.py:77]: Can you move this `request.get()` to the client and have the client send an `Image.Image` instead? Ideally, we'd like to avoid getting remote urls in these inference methods.

File: examples/stable-video-diffusion/models/svd.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            self.cfg.model_name,
+            torch_dtype=self.torch_dtype,
+            variant=self.variant
+        )
+
+        self.pipe.enable_model_cpu_offload()
+
+    def image2video(
+        self,
+        image: Union[np.ndarray, Image.Image, str],
```
Comment (spillai) [inline examples/stable-video-diffusion/models/svd.py:69]: Based on comment below, only support `Union[np.ndarray, Image.Image]`

File: examples/stable-video-diffusion/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+``` python
+from PIL import Image
+from nos.client import Client
+
+client = Client("[::]:50051")
+model = client.Module("stable-video-diffusion")
+
+image_link = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/svd/rocket.png?download=true"
+
+response: Iterable[Image.Image] = model.image2video(image=image_link, _stream=True)
```
Comment (spillai) [inline examples/stable-video-diffusion/README.md:22]: Do a `image: Image.Image = request.get(...)` here and pass the image instead to the `image2video` method.

File: examples/stable-video-diffusion/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+frames = []
```
Comment (spillai) [inline examples/stable-video-diffusion/README.md:23]: `frames = list(response)` should iterate and create the list of frames.

File: examples/stable-video-diffusion/models/svd.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+import numpy as np
+import torch
+
+from nos.hub import HuggingFaceHubConfig
+
+@dataclass(frozen=True)
+class StableVideoDiffusionConfig(HuggingFaceHubConfig):
+    
+    model_cls: str = "StableVideoDiffusionPipeline"
+
```
Comment (spillai) [inline examples/stable-video-diffusion/models/svd.py:17]: Add docstrings if possible for each field. See https://github.com/autonomi-ai/nos/blob/main/nos/models/stable_diffusion.py#L14 for reference.

File: examples/stable-video-diffusion/models/svd.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        if isinstance(image, str):
+            if image.startswith("http://") or image.startswith("https://"):
+                image = Image.open(requests.get(image, stream=True).raw)
+            elif os.path.isfile(image):
+                image = Image.open(image)
+
+        image = image.resize((self.cfg.width, self.cfg.height))
+
+        generator = torch.manual_seed(seed)
+        frames = self.pipe(image, decode_chunk_size=self.cfg.decode_chunk_size, generator=generator).frames[0]
```
Comment (spillai) [inline examples/stable-video-diffusion/models/svd.py:85]: You can instead use `yield from self.pipe(...)` and avoid writing the remaining two lines.

File: examples/stable-video-diffusion/models/svd.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+        for frame in frames:
```
Comment (spillai) [inline examples/stable-video-diffusion/models/svd.py:87]: Can remove this if `yield from` is used.

File: examples/stable-video-diffusion/tests/test_svd.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    model_id = "stable-video-diffusion"
+    models: List[str] = client.ListModels()
+    assert model_id in models
+
+    model = client.Module(model_id)
+    assert model is not None
+    assert model.GetModelInfo() is not None
+    print(f"Test [model={model_id}]")
+
+    response: Iterable[Image.Image] = model.image2video(image=TEST_IMAGE_LINK, _stream=True)
```
Comment (spillai) [inline examples/stable-video-diffusion/tests/test_svd.py:24]: Same comment as above, would be good to pass in Image.Image instead of a link. But otherwise, looks great!

File: examples/stable-video-diffusion/tests/test_svd.py [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    frames = []
```
Comment (spillai) [inline examples/stable-video-diffusion/tests/test_svd.py:25]: frames = list(response)

**Review by spillai — CHANGES_REQUESTED** [review]
Left a few changes, 💯

File: examples/stable-video-diffusion/models/svd.py (hunk 1 above)
Comment (jiexiong2016) [reply examples/stable-video-diffusion/models/svd.py:77]: Got it, make sense. Removed str as input.

File: examples/stable-video-diffusion/models/svd.py (hunk 2 above)
Comment (jiexiong2016) [reply examples/stable-video-diffusion/models/svd.py:69]: Got it, make sense. Removed str as input.

File: examples/stable-video-diffusion/README.md (hunk 3 above)
Comment (jiexiong2016) [reply examples/stable-video-diffusion/README.md:22]: Got it, changed.

File: examples/stable-video-diffusion/README.md (hunk 4 above)
Comment (jiexiong2016) [reply examples/stable-video-diffusion/README.md:23]: Got it, changed.

File: examples/stable-video-diffusion/tests/test_svd.py (hunk 8 above)
Comment (jiexiong2016) [reply examples/stable-video-diffusion/tests/test_svd.py]: Got it, changed.

File: examples/stable-video-diffusion/tests/test_svd.py (hunk 9 above)
Comment (jiexiong2016) [reply examples/stable-video-diffusion/tests/test_svd.py]: Removed frames here since it is not used.

File: examples/stable-video-diffusion/models/svd.py (hunk 7 above)
Comment (jiexiong2016) [reply examples/stable-video-diffusion/models/svd.py:77]: True, changed.

File: examples/stable-video-diffusion/models/svd.py (hunk 5 above)
Comment (jiexiong2016) [reply examples/stable-video-diffusion/models/svd.py:15]: Done.

**Comment by jiexiong2016** [comment]
Rebased.

**Review by spillai — APPROVED** [review]
